### PR TITLE
Enable traces filter configuration from the shake detector.

### LINKED ISSUE #15: Configure traces filter from the shake detector.
I'm using lynx and it's open automatically if I shake the phone. The problem with this usage is that right now it's impossible to configure the filter we are going to use by default. 

We should be able to pass a LynxConfig instance to the ShakeDetector. This config object will be used to start LynxActivity when the user shakes the phone.

This is the current Lynx usage from any Application class:

```java

public void onCreate(){
    super.onCreate();
    initializeLynx();
}

private void initializeLynx() {
    if (BuildConfig.DEBUG) {
      LynxShakeDetector lynxShakeDetector = new LynxShakeDetector(this);
      lynxShakeDetector.init();
    }
  }

```

This is the proposed configuration:

```java

public void onCreate(){
    super.onCreate();
    initializeLynx();
}

private void initializeLynx() {
    if (BuildConfig.DEBUG) {
      LynxShakeDetector lynxShakeDetector = new LynxShakeDetector(this);
        LynxConfig lynxConfig = new LynxConfig();
        lynxConfig.setMaxNumberOfTracesToShow(4000).setFilter("LynxFilter");
      lynxShakeDetector.init(lynxConfig);
    }
  }

```

### PR DESCRIPTION
**Pull request by vic-gonzi**
Fix #15

### CODE REVIEW

**Comment by vic-gonzi** [comment]
@pedrovgs It seems Circle servers didn't update Nougat SDK:

`failed to find target with hash string 'android-24' in: /usr/local/android-sdk-linux
`
Everything else is passing, so you may consider waiving it, or else to wait for an internal Circle update.

**Comment by pedrovgs** [comment]
Thank's for your PR @vic-gonzi :)

Don't worry about Circle-CI, I completely forgot removing that github check. In this repository we are using just Travis-CI. Your build is passing perfectly. I've removed all the Circle-CI hook.

File: sample/src/main/java/com/github/pedrovgs/sample/LynxApplication.java [hunk 1]
```diff
@@ -30,6 +31,8 @@
   @Override public void onCreate() {
     super.onCreate();
     LynxShakeDetector lynxShakeDetector = new LynxShakeDetector(this);
+    LynxConfig lynxConfig = new LynxConfig();
+    lynxConfig.setMaxNumberOfTracesToShow(4000).setFilter("LynxFilter");
     lynxShakeDetector.init();
```
Comment (pedrovgs) [inline sample/src/main/java/com/github/pedrovgs/sample/LynxApplication.java]: Thanks for the sample update :) But don't we forget to pass the new ``lynxConfig`` instance to the ``lynxShakeDetector.init()`` call?

**Comment by vic-gonzi** [comment]
@pedrovgs Same issue than here: https://github.com/pedrovgs/Lynx/pull/21 . You may want to put this PR on hold as well.

**Comment by pedrovgs** [comment]
Keeping a PR on hold until Travis is updated is something we can not do is we want to publish a new release with this code. Instead of block this PR why don't you update the ``.travis.yml`` configuration to use that build tools version and if this version is not available uses the build config version available in Travis-CI environments?

**Comment by vic-gonzi** [comment]
@pedrovgs Build tools version v24.0.0 seems to be fixed now from Travis side. This branch should not need any further change, as both build.gradle and .travis.yml refer to 24.0.0 . Just rerun Travis, and we should get a green light.

**Comment by pedrovgs** [comment]
👍

**Pull request merged**